File: 프로그래머스/unrated/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
//1. 기본 풀이
// class Solution {
//     public int[] solution(int[] arr, int n) {
//         if(arr.length % 2 == 0) {
//             for(int i = 1; i < arr.length; i += 2)
//                 arr[i] += n;
//         }else{
//             for(int i = 0; i < arr.length; i += 2)
//                 arr[i] += n;
//         }
//         return arr;
//     }
// }

//2. 스트림 풀이
import java.util.stream.IntStream;

class Solution {
    public int[] solution(int[] arr, int n) {
        return IntStream.range(0, arr.length)
                        .map(i -> i % 2 == 0 && arr.length % 2 == 1 || i % 2 == 1 && arr.length % 2 == 0 ? arr[i] + n : arr[i])
                        .toArray();
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int n) {
    bool order = arr.size() % 2 == 0;
    int i = order ? 1 : 0;
    while(i < arr.size()){
        arr[i] += n;
        i += 2;
}
    return arr;
}